flight/PiOS/Common/pios_mpu9250_spi.c

```diff
@@ -352,8 +353,16 @@ static int32_t PIOS_MPU9250_Config(struct pios_mpu9250_cfg const *cfg)
 	PIOS_DELAY_WaitmS(50);
 
 	uint8_t id = PIOS_MPU9250_ReadReg(PIOS_MPU60X0_WHOAMI);
-	if (id != MPU9250_WHOAMI_ID)
-		return -2;
+	switch (id) {
```

Comment:
How about pulling these changes out into a separate PR? Combining driver changes with new board support makes tracking changes harder than it should be.